project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.ts

```diff
@@ -75,16 +158,92 @@ export class UserCardComponent implements OnInit {
       })
   }
 
-  editUser(user: any) {
-    this.usersData.content.forEach((u: any) => {
-      if (u.userId === user.userId && user.editUser === false) {
-        u.editUser = !u.editUser
+  async loadGroups() {
+    await this.usersSvc.getGroups().subscribe(
+      (data: any) => {
+        const res = data.result.response
+        res.map((value: any) => {
+          this.groupsList.push({ name: value })
+        })
+      },
+      (_err: any) => {
+      })
+  }
+
+  async loadLangauages() {
+    await this.usersSvc.getMasterLanguages().subscribe(
+      (data: any) => {
+        this.masterLanguagesEntries = data.languages
+        this.onChangesLanuage()
+      },
+      (_err: any) => {
+      })
+  }
+
+  async loadCountryCodes() {
+    this.usersSvc.getMasterNationlity().subscribe((data: any) => {
+      data.nationality.map((item: any) => {
+        this.countryCodes.push(item.countryCode)
+      })
+
+      this.updateUserDataForm.patchValue({
+        countryCode: '+91',
+      })
+    },
+                                                  (_err: any) => {
+      })
+  }
+
+  closeOtherPanels(openPanel: MatExpansionPanel) {
+    this.panels.forEach(panel => {
+      if (panel !== openPanel) {
+        panel.close()
       }
     })
   }
 
+  otherDropDownChange(value: any, field: string) {
+    if (field === 'designation' && value !== 'Other') {
+      this.updateUserDataForm.controls['designation'].setValue(value)
+    }
+  }
+
+  onChangesLanuage(): void {
+    // tslint:disable-next-line: no-non-null-assertion
+    this.masterLanguages = this.updateUserDataForm.get('domicileMedium')!.valueChanges
+      .pipe(
+        debounceTime(500),
+        distinctUntilChanged(),
+        startWith(''),
+        map((value: any) => typeof (value) === 'string' ? value : (value && value.name ? value.name : '')),
+        map((name: any) => name ? this.filterLanguage(name) : this.masterLanguagesEntries.slice()),
+      )
+  }
+
+  private filterLanguage(name: string) {
+    if (name) {
+      const filterValue = name.toLowerCase()
+      return this.masterLanguagesEntries.filter((option: any) => option.name.toLowerCase().includes(filterValue))
+    }
+    return this.masterLanguagesEntries
+  }
+
+  numericOnly(event: any): boolean {
+    const pattren = /^([0-9])$/
```

Comment:
please use pattern as a variable

Reply:
It s added as a event function check on paste & keypress